components/admin/permit-holders/modals/GenerateReportModal.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  Button,
+  Text,
+  Box,
+  Stack,
+  Input,
+  FormControl,
+  FormLabel,
+  Checkbox,
+  SimpleGrid,
+  ListItem,
+} from '@chakra-ui/react'; // Chakra UI
+import { DownloadIcon } from '@chakra-ui/icons';
+import { useState } from 'react'; // React
+
+type GenerateReportProps = {
+  readonly isOpen: boolean;
+  readonly onClose: () => void;
+};
+
+enum GenerateReportModalEnum {
+  SELECT_COLUMNS = 1,
+  EXPORT = 2,
+}
+
+export default function GenerateReportModal(props: GenerateReportProps) {
+  const { isOpen, onClose } = props;
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
+  const [step, setStep] = useState(GenerateReportModalEnum.SELECT_COLUMNS);
+
+  if (step == 1) {
+    return (
+      <>
+        <Modal isOpen={isOpen} onClose={onClose} isCentered size={'xl'}>
```

Comment:
We don't want each step to render a new `Modal` instance. Instead, what we can do is return a `Modal` and make a function `renderStep` which takes in a `step` parameter (of the enum type), and the `renderStep` function returns the content for the modal step.